app/models/concerns/audit_extensions.rb

```diff
@@ -33,9 +34,27 @@ def ensure_taxonomies_not_escalated
   private
 
   def log_audit
-    Foreman::Logging.with_fields(self.audited_changes) do
-      Foreman::Logging.logger('audit').info { "#{self.action} event for #{self.auditable_type} with id #{self.auditable_id}" }
+    telemetry_increment_counter(:audit_records_created, 1, type: self.auditable_type)
+    aulogger = Foreman::Logging.logger('audit')
+    return unless (self.audited_changes && aulogger.info?)
+    self.audited_changes.each_pair do |column_name, old_and_new|
+      audited_fields = {
+        audit_action: self.action,
+        audit_type: self.auditable_type,
+        audit_id: self.auditable_id,
+        audit_column: column_name
+      }
+      if old_and_new.is_a?(Array)
+        audited_fields[:audit_field_old] = old_and_new[0]
+        audited_fields[:audit_field_new] = old_and_new[1]
+      else
+        audited_fields[:audit_field] = old_and_new
+      end
+      Foreman::Logging.with_fields(audited_fields) do
+        aulogger.info "#{self.auditable_type} (#{self.auditable_id}) #{self.action} event on #{column_name} #{old_and_new}"
```

Comment:
please add some separator between column name and the value, e.g. dash or colon, it's hard to read, see example below

```
[I|aud|c5982] Subnet (13) create event on network 192.168.25.0
[I|aud|c5982] Subnet (13) create event on mask 255.255.255.0
[I|aud|c5982] Subnet (13) create event on priority 
[I|aud|c5982] Subnet (13) create event on name delete
[I|aud|c5982] Subnet (13) create event on vlanid 
[I|aud|c5982] Subnet (13) create event on dhcp_id 
[I|aud|c5982] Subnet (13) create event on tftp_id 
[I|aud|c5982] Subnet (13) create event on gateway 
[I|aud|c5982] Subnet (13) create event on dns_primary 
[I|aud|c5982] Subnet (13) create event on dns_secondary 
[I|aud|c5982] Subnet (13) create event on from 
[I|aud|c5982] Subnet (13) create event on to 
[I|aud|c5982] Subnet (13) create event on dns_id 
[I|aud|c5982] Subnet (13) create event on boot_mode DHCP
[I|aud|c5982] Subnet (13) create event on ipam DHCP
[I|aud|c5982] Subnet (13) create event on discovery_id 
[I|aud|c5982] Subnet (13) create event on description 
[I|aud|c5982] Subnet (13) create event on mtu 1500
[I|aud|c5982] Subnet (13) create event on template_id 
[I|aud|c5982] Subnet (13) create event on organizations Abcde, Ares, Demo Org
[I|aud|c5982] Subnet (13) create event on locations Brno
```